### ISSUE TITLE
Bisection error with Unitful

### ISSUE THREAD

**Issue by tomyun** [JuliaMath/Roots.jl#162]
```julia
using Roots
using Unitful
s = u"s"; m = u"m"
g = 9.8*m/s^2
v0 = 10m/s
y0 = 16m
y(t) = -g*t^2 + v0*t + y0
find_zero(y, 1s)
# 1.8860533706680143 s
```

Above example in README works as is. However, it fails when bisection method is used.

```julia
find_zero(y, (0s, 2s), Bisection())
# ERROR: DimensionError: 0.0 and 0.0 s are not dimensionally compatible.
```

Interestingly other methods work fine with the same bracket.

```julia
find_zero(y, (0s, 2s), FalsePosition())
find_zero(y, (0s, 2s), Roots.A42())
find_zero(y, (0s, 2s), Roots.AlefeldPotraShi())
# 1.8860533706680243 s
```

I guess it should have been caught in the [test case](https://github.com/JuliaMath/Roots.jl/blob/95a79efd71a23011268723e5a60517e07b9c62f3/test/test_composable.jl#L31) if it was not [commented out](https://github.com/JuliaMath/Roots.jl/blob/95a79efd71a23011268723e5a60517e07b9c62f3/test/runtests.jl#L14).

**Comment by jverzani** [comment]
Thanks for letting me know. I just pushed a fix to address this bug. But still should add `Unitful` to the test suite so that those tests run automatically.

**Comment by jverzani** [comment]
closed by #163
